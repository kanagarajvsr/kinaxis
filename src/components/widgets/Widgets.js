import "./widgets.scss";
import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp"; 
import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown"; 
import { Link } from "react-router-dom";

const Widget = ({ data }) => {
  return (
    <div className="widget">
      <div className="left">
        <span className="title">{data.title}</span>
        <span className="counter">
          {data.isMoney && "Rs:"} {data.amount}
        </span>
          <span className="link"><Link to={data.redirect}>{data.link}</Link></span>
      </div>
      <div className="right">
        <div className={`percentage ${data.diff<0? 'negative':'positive'}`}>
        {
          data.diff>0?<KeyboardArrowUpIcon />:<KeyboardArrowDownIcon />
        }
          {data.diff} %
        </div>
        {data.icon}
      </div>
    </div>
  );
};

export default Widget;
