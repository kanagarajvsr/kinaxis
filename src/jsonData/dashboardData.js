import PersonOutlinedIcon from "@mui/icons-material/PersonOutlined";
import AccountBalanceWalletOutlinedIcon from "@mui/icons-material/AccountBalanceWalletOutlined";
import ShoppingCartOutlinedIcon from "@mui/icons-material/ShoppingCartOutlined";
import MonetizationOnOutlinedIcon from "@mui/icons-material/MonetizationOnOutlined";

export const charts =[
    { name: "January", Total: 1200 },
    { name: "February", Total: 2100 },
    { name: "March", Total: 800 },
    { name: "April", Total: 1600 },
    { name: "May", Total: 900 },
    { name: "June", Total: 1700 },
    { name: "July", Total: 600 },
    { name: "Augest", Total: 900 },
    { name: "September", Total: 1400 },
    { name: "October", Total: 1200 },
    { name: "November", Total: 1600 },
    { name: "December", Total: 1100 }
  ];

export const widgets={
    user:{
      title: "USERS",
      isMoney: false,
      link: "See all users",
      redirect: "/users",
      amount:200,
      diff:10,
      icon: (
        <PersonOutlinedIcon
          className="icon"
          style={{
            color: "crimson",
            backgroundColor: "rgba(255, 0, 0, 0.2)",
          }}
        />
      ),
    },
     order: {
      title: "ORDERS",
      isMoney: false,
      link: "View all orders",
      redirect: "/orders",
      amount:300,
      diff:30,
      icon: (
        <ShoppingCartOutlinedIcon
          className="icon"
          style={{
            backgroundColor: "rgba(218, 165, 32, 0.2)",
            color: "goldenrod",
          }}
        />
      ),
    },
    earning: {
      title: "EARNINGS",
      isMoney: true,
      link: "View net earnings",
      redirect: "/earnings",
      amount:1000,
      diff:-5,
      icon: (
        <MonetizationOnOutlinedIcon
          className="icon"
          style={{ backgroundColor: "rgba(0, 128, 0, 0.2)", color: "green" }}
        />
      ),
    },
    balance: {
      title: "BALANCE",
      isMoney: true,
      link: "See details",
      redirect: "/transactions",
      amount:2000,
      diff:20,
      icon: (
        <AccountBalanceWalletOutlinedIcon
          className="icon"
          style={{
            backgroundColor: "rgba(128, 0, 128, 0.2)",
            color: "purple",
          }}
        />
      ),
    }
}