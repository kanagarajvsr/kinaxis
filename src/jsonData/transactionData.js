 
export const Columns = [
    { field: "id", headerName: "Tracking ID", width: 100 },
    { field: "product", headerName: "Product Name", width: 220 },
    {
      field: "customer",
      headerName: "Customer",
      width: 220
    },
    {
      field: "date",
      headerName: "Date",
      width: 100,
    },
  
    {
      field: "amount",
      headerName: "Amount",
      width: 150,
    },
    {
      field: "method",
      headerName: "Payment Method",
      width: 150,
    },
    {
      field: "status",
      headerName: "Status",
      width: 80,
      renderCell: (params) => {
        return (
          <div className={`cellWithStatus ${params.row.status}`}>
            {params.row.status}
          </div>
        );
      },
    },
  ];
  
 
    export const Rows = [
      {
        id: 1143155,
        product: "Acer Nitro 5",
        customer: "John Smith",
        date: "1 March",
        amount: 785,
        method: "Cash on Delivery",
        status: "Approved",
      },
      {
        id: 2235235,
        product: "Playstation 5",
        customer: "Michael Doe",
        date: "1 March",
        amount: 900,
        method: "Online Payment",
        status: "Pending",
      },
      {
        id: 2342353,
        product: "Redragon S101",
        customer: "John Smith",
        date: "1 March",
        amount: 35,
        method: "Cash on Delivery",
        status: "Pending",
      },
      {
        id: 2357741,
        product: "Razer Blade 15",
        customer: "Jane Smith",
        date: "1 March",
        amount: 920,
        method: "Online",
        status: "Approved",
      },
      {
        id: 2342355,
        product: "ASUS ROG Strix",
        customer: "Harold Carol",
        date: "1 March",
        amount: 2000,
        method: "Online",
        status: "Pending",
      },
    ];