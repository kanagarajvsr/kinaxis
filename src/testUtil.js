
import { BrowserRouter } from 'react-router-dom'
import { render,screen,fireEvent } from '@testing-library/react';
 
export const renderWithRouter = (ui, {route = '/'} = {}) => {
    window.history.pushState({}, 'Test page', route);
    const renders = render(ui, {wrapper: BrowserRouter})
    return { 
        screen,
        fireEvent,
        ...renders
    }
  }