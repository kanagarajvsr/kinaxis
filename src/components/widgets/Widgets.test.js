import Widgets from './Widgets';
import {renderWithRouter} from '../../testUtil';
import { widgets  } from "../../jsonData/dashboardData";
import { cleanup } from '@testing-library/react';

describe('Widgets render', () => {
  afterEach(cleanup);

  test('Update Snapshot with users', () => { 
      const utils =  renderWithRouter(<Widgets data={ widgets.user} />)
      const {asFragment}=utils;
      expect(asFragment()).toMatchSnapshot()
  })

  test('Update Snapshot with order', () => { 
      const utils =  renderWithRouter(<Widgets data={ widgets.order} />)
      const {asFragment}=utils;
      expect(asFragment()).toMatchSnapshot()
  })

  test('Update Snapshot with earning', () => { 
      const utils =  renderWithRouter(<Widgets data={ widgets.earning} />)
      const {asFragment}=utils;
      expect(asFragment()).toMatchSnapshot()
  })

  test('Update Snapshot with balance', () => { 
      const utils =  renderWithRouter(<Widgets data={ widgets.balance} />)
      const {asFragment}=utils;
      expect(asFragment()).toMatchSnapshot()
  })
  
  });
   


