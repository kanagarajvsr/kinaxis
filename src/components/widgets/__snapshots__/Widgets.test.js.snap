// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets render Update Snapshot with balance 1`] = `
<DocumentFragment>
  <div
    class="widget"
  >
    <div
      class="left"
    >
      <span
        class="title"
      >
        BALANCE
      </span>
      <span
        class="counter"
      >
        Rs: 2000
      </span>
      <span
        class="link"
      >
        <a
          href="/transactions"
        >
          See details
        </a>
      </span>
    </div>
    <div
      class="right"
    >
      <div
        class="percentage positive"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowUpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
          />
        </svg>
        20 %
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
        data-testid="AccountBalanceWalletOutlinedIcon"
        focusable="false"
        style="background-color: rgba(128, 0, 128, 0.2); color: purple;"
        viewBox="0 0 24 24"
      >
        <path
          d="M21 7.28V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-2.28c.59-.35 1-.98 1-1.72V9c0-.74-.41-1.37-1-1.72zM20 9v6h-7V9h7zM5 19V5h14v2h-6c-1.1 0-2 .9-2 2v6c0 1.1.9 2 2 2h6v2H5z"
        />
        <circle
          cx="16"
          cy="12"
          r="1.5"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Widgets render Update Snapshot with earning 1`] = `
<DocumentFragment>
  <div
    class="widget"
  >
    <div
      class="left"
    >
      <span
        class="title"
      >
        EARNINGS
      </span>
      <span
        class="counter"
      >
        Rs: 1000
      </span>
      <span
        class="link"
      >
        <a
          href="/earnings"
        >
          View net earnings
        </a>
      </span>
    </div>
    <div
      class="right"
    >
      <div
        class="percentage negative"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
          />
        </svg>
        -5 %
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
        data-testid="MonetizationOnOutlinedIcon"
        focusable="false"
        style="background-color: rgba(0, 128, 0, 0.2); color: green;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm.31-8.86c-1.77-.45-2.34-.94-2.34-1.67 0-.84.79-1.43 2.1-1.43 1.38 0 1.9.66 1.94 1.64h1.71c-.05-1.34-.87-2.57-2.49-2.97V5H10.9v1.69c-1.51.32-2.72 1.3-2.72 2.81 0 1.79 1.49 2.69 3.66 3.21 1.95.46 2.34 1.15 2.34 1.87 0 .53-.39 1.39-2.1 1.39-1.6 0-2.23-.72-2.32-1.64H8.04c.1 1.7 1.36 2.66 2.86 2.97V19h2.34v-1.67c1.52-.29 2.72-1.16 2.73-2.77-.01-2.2-1.9-2.96-3.66-3.42z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Widgets render Update Snapshot with order 1`] = `
<DocumentFragment>
  <div
    class="widget"
  >
    <div
      class="left"
    >
      <span
        class="title"
      >
        ORDERS
      </span>
      <span
        class="counter"
      >
         300
      </span>
      <span
        class="link"
      >
        <a
          href="/orders"
        >
          View all orders
        </a>
      </span>
    </div>
    <div
      class="right"
    >
      <div
        class="percentage positive"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowUpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
          />
        </svg>
        30 %
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
        data-testid="ShoppingCartOutlinedIcon"
        focusable="false"
        style="background-color: rgba(218, 165, 32, 0.2); color: goldenrod;"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.37-.66-.11-1.48-.87-1.48H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Widgets render Update Snapshot with users 1`] = `
<DocumentFragment>
  <div
    class="widget"
  >
    <div
      class="left"
    >
      <span
        class="title"
      >
        USERS
      </span>
      <span
        class="counter"
      >
         200
      </span>
      <span
        class="link"
      >
        <a
          href="/users"
        >
          See all users
        </a>
      </span>
    </div>
    <div
      class="right"
    >
      <div
        class="percentage positive"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="KeyboardArrowUpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
          />
        </svg>
        10 %
      </div>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
        data-testid="PersonOutlinedIcon"
        focusable="false"
        style="color: crimson; background-color: rgba(255, 0, 0, 0.2);"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 6c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m0 10c2.7 0 5.8 1.29 6 2H6c.23-.72 3.31-2 6-2m0-12C9.79 4 8 5.79 8 8s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 10c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
